File: codeK(main)/Top 150 Interview Questions/0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int k = 0;
        for(int i = 0; i < nums.length; i++){
        if(nums[i] != nums[k]){
            k++;
            nums[k] = nums[i];
        }
        }
        return k + 1;
    }
}

File: codeK(main)/Leetcode/0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    if (nums.length === 0) {
        return 0
    }

    let k = 1

    for (let i = 1; i<nums.length; i++){
        if(nums[i] !== nums[i - 1]) {
            nums[k] = nums[i]
            k++
        }
    }
    return k
};

// trick learned: 2 pointers, consider the first element as unique